src/pages/users/Users.tsx

```diff
@@ -26,12 +36,70 @@ enum UserRole {
   Facilitator = "Facilitator",
 }
 
+const defaultUserData: UserRequest = {
+  email: "",
+  is_admin: false,
+};
+
 const Users = () => {
   const classes = useStyles();
   const { users } = useContext(UsersContext);
+  const [displayInviteDialog, setIsDisplayInviteDialog] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState("");
+  const [newUser, setNewUser] = useState(defaultUserData);
+
+  const handleOpenInviteDialog = () => {
+    setIsDisplayInviteDialog(true);
+  };
+
+  const handleCloseInviteDialog = () => {
+    setNewUser(defaultUserData);
+    setIsDisplayInviteDialog(false);
+  };
+
+  const convertBooleanToRole = (isAdmin: boolean) =>
+    isAdmin ? UserRole.Admin : UserRole.Facilitator;
+  const convertRoleToBoolean = (role: UserRole) => role === UserRole.Admin;
+
+  const inviteUser = async () => {
+    try {
+      await UserAPI.postUser(newUser);
+      handleCloseInviteDialog();
+      setSnackbarMessage(
+        `Successfully added ${newUser.email} to the application.`
+      );
```

Comment:
we'll also want to call [`fetchUsers`](https://github.com/uwblueprint/project-read-frontend/blob/f1b3106eba8d230522ff5d01ba8551c85668e6bf/src/context/UsersContext.tsx#L25) so that the users page displays the new user, as well as in the staff selects in the rest of the app :)

this will need a tiny update to the users context, check out the [`DynamicFieldsContext`](https://github.com/uwblueprint/project-read-frontend/blob/f1b3106eba8d230522ff5d01ba8551c85668e6bf/src/context/DynamicFieldsContext.tsx#L39) and how we refactored out `fetchDynamicFields` there!